crates/cli-support/src/js/binding.rs

```diff
@@ -1291,85 +1311,71 @@ fn instruction(
-                    i = i,
-                    args = args,
-                    wrapper = wrapper,
-                ));
-            }
+            let cb = js.alias(
+                "cb",
+                if *mutable {
+                    // Mutable closures need protection against being called
+                    // recursively, so ensure that we clear out one of the
+                    // internal pointers while it's being invoked.
+                    format!(
+                        "({args}) => {{
+                            const a = {state}.a;
+                            {state}.a = 0;
+                            try {{
+                                return {wrapper}(a, {state}.b, {args});
+                            }} finally {{
+                                {state}.a = a;
+                            }}
+                        }}"
+                    )
+                } else {
+                    format!("({args}) => {wrapper}({state}.a, {state}.b, {args})")
+                },
+            );
 
             // Make sure to null out our internal pointers when we return
             // back to Rust to ensure that any lingering references to the
             // closure will fail immediately due to null pointers passed in
             // to Rust.
-            js.finally(&format!("state{}.a = state{0}.b = 0;", i));
-            js.push(format!("cb{}", i));
+            js.finally(&format!("{state}.a = {state}.b = 0;"));
+            js.push(cb);
         }
 
         Instruction::VectorLoad { kind, mem, free } => {
             let len = js.pop();
             let ptr = js.pop();
             let f = js.cx.expose_get_vector_from_wasm(kind.clone(), *mem)?;
-            let i = js.tmp();
             let free = js.cx.export_name_of(*free);
-            js.prelude(&format!("var v{} = {}({}, {}).slice();", i, f, ptr, len));
+            let val = js.alias("v", format!("{f}({ptr}, {len}).slice()"));
             js.prelude(&format!(
-                "wasm.{}({}, {} * {size}, {size});",
-                free,
-                ptr,
-                len,
+                "wasm.{free}({ptr}, {len} * {size}, {size});",
                 size = kind.size()
             ));
-            js.push(format!("v{}", i))
+            js.push(val);
         }
 
         Instruction::OptionVectorLoad { kind, mem, free } => {
             let len = js.pop();
             let ptr = js.pop();
             let f = js.cx.expose_get_vector_from_wasm(kind.clone(), *mem)?;
-            let i = js.tmp();
             let free = js.cx.export_name_of(*free);
-            js.prelude(&format!("let v{};", i));
+            let val = js.r#let("v", "undefined".to_string());
             js.prelude(&format!("if ({} !== 0) {{", ptr));
-            js.prelude(&format!("v{} = {}({}, {}).slice();", i, f, ptr, len));
+            js.prelude(&format!("{val} = {f}({ptr}, {len}).slice();"));
```

Comment:
Shouldn't we just be able to use `var` here?

Reply:
With the introduction of `let`, the purpose of `var` has now become to declare variables as accessible from within the `finally` block.

The goal is to generate correct JS code, not code golfing.

Reply:
In lack of a proper response: please use `var` here.